gateway/api/v1/serializers.py

```diff
@@ -43,6 +45,30 @@ def validate(self, attrs):
                 "At least one of attributes (entrypoint, image) is required."
             )
 
+        # validate dependencies
+        # allowlist stored in json config file (eventually via configmap)
+        # sample:
+        # allowlist = { "wheel": ["0.44.0", "0.43.2"] }
+        # where the values for each key are allowed versions of dependency
+        deps = json.loads(attrs.get("dependencies", None))
+        with open(settings.GATEWAY_ALLOWLIST_CONFIG, encoding="utf-8", mode="r") as f:
+            allowlist = json.load(f)
+
+        # If no allowlist specified, all dependencies allowed
+        if len(allowlist.keys()) > 0:
```

Comment:
should we check first is `allowlist` is `None`?

Reply:
yeah, should catch possible errors here